cpp/src/parquet/arrow/reader.h

```diff
@@ -151,8 +151,10 @@ class PARQUET_EXPORT FileReader {
       int i, std::shared_ptr<::arrow::ChunkedArray>* out) = 0;
 
   /// \brief Return a RecordBatchReader of row groups selected from row_group_indices, the
-  ///    ordering in row_group_indices matters.
-  /// \returns error Status if row_group_indices contains invalid index
+  ///     ordering in row_group_indices matters. Each row group will held in memory until
+  ///     each slice has been yielded as a RecordBatch, at which point the next row group
+  ///     will be loaded. FileReaders must outlive their RecordBatchReaders.
```

Comment:
Probably want to add a TODO here that we want to fix this (this is a problem for memory use)